`TestIntegrations/X11Forwarding` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/11442800048/job/31833983642

#### Relevant snippet


```
==================
WARNING: DATA RACE
Write at 0x00c005069900 by goroutine 56906:
  github.com/gravitational/teleport/integration.testX11Forwarding.func3.3.3.1()
      /__w/teleport/teleport/integration/integration_test.go:4752 +0xa4
  github.com/stretchr/testify/assert.Eventually.func1()
      /go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x33

Previous write at 0x00c005069900 by goroutine 56878:
  github.com/gravitational/teleport/integration.testX11Forwarding.func3.3.3.1()
      /__w/teleport/teleport/integration/integration_test.go:4752 +0xa4
  github.com/stretchr/testify/assert.Eventually.func1()
      /go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x33

Goroutine 56906 (running) created at:
  github.com/stretchr/testify/assert.Eventually()
      /go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x3d5
  github.com/gravitational/teleport/integration.testX11Forwarding.func3.3.3()
      /__w/teleport/teleport/integration/integration_test.go:4749 +0x31e
  github.com/stretchr/testify/assert.EventuallyWithT.func1()
      /go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1983 +0xb7

Goroutine 56878 (finished) created at:
  github.com/stretchr/testify/assert.Eventually()
      /go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x3d5
  github.com/gravitational/teleport/integration.testX11Forwarding.func3.3.3()
      /__w/teleport/teleport/integration/integration_test.go:4749 +0x31e
  github.com/stretchr/testify/assert.EventuallyWithT.func1()
      /go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1983 +0xb7
==================
```

Looks like the data race is gone but still seeing failures:
```
    integration_test.go:4744: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:4749
        	            				/opt/go/src/runtime/asm_amd64.s:1700
        	Error:      	Condition never satisfied
        	Messages:   	failed to read display
    integration_test.go:4744: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:4744
        	Error:      	Condition never satisfied
        	Test:       	TestIntegrations/X11Forwarding/recording_mode=node/controlMaster=false
            --- FAIL: TestIntegrations/X11Forwarding/recording_mode=node/controlMaster=false (11.40s)
```

--- a/integration/integration_test.go
+++ b/integration/integration_test.go
@@ -4853,12 +4853,15 @@ func testX11Forwarding(t *testing.T, suite *integrationTestSuite) {
 							assert.Eventually(t, func() bool {
 								output, err := os.ReadFile(tmpFile.Name())
 								if err == nil && len(output) != 0 {
-									display <- strings.TrimSpace(string(output))
+									select {
+									case display <- strings.TrimSpace(string(output)):
+									default:
+									}
 									return true
 								}
 								return false
 							}, time.Second, 100*time.Millisecond, "failed to read display")
-						}, 10*time.Second, 2*time.Second)
+						}, 10*time.Second, 1*time.Second)
 
 						// Make a new connection to the XServer proxy to confirm that forwarding is working.
 						serverDisplay, err := x11.ParseDisplay(<-display)

--- a/integration/integration_test.go
+++ b/integration/integration_test.go
@@ -4858,7 +4858,7 @@ func testX11Forwarding(t *testing.T, suite *integrationTestSuite) {
 								}
 								return false
 							}, time.Second, 100*time.Millisecond, "failed to read display")
-						}, 10*time.Second, time.Second)
+						}, 10*time.Second, 2*time.Second)
 
 						// Make a new connection to the XServer proxy to confirm that forwarding is working.
 						serverDisplay, err := x11.ParseDisplay(<-display)

